packages/chevrotain/src/scan/lexer_public.ts

```diff
@@ -690,7 +694,12 @@ export class Lexer {
         const errorLine = line
         const errorColumn = column
         let foundResyncPoint = false
-        while (!foundResyncPoint && offset < orgLength) {
+
+        if (!lexerConfig.recoveryEnabled) {
+          failedBeforeEnd = true
+        }
+
+        while (!failedBeforeEnd && !foundResyncPoint && offset < orgLength) {
```

Comment:
Suggestion: Instead of checking `!failedBeforeEnd` during each run of this loop and the one above, you could probably just replace this block with the one below and add an `if (!recoveryEnabled) break` statement after pushing the lexer error. That way we ensure that no change affects the common lexer path (in which no lexer errors appear) and also remove the need for the `failedBeforeEnd` variable as a whole.

By the way, this is just a really minor safe guard against possible performance regressions. I couldn't identify a statistically significant performance regression with your changes.

```ts
        let foundResyncPoint = !recoveryEnabled
        while (!foundResyncPoint && offset < orgLength) {
```

Reply:
Ah yes, that is nicer! Glad it didn't impact performance noticeably though - it's really only in the already-edge case that no token matches anyway.

New commit with improved implementation.